src/annotator/sidebar.js

```diff
@@ -27,10 +28,34 @@ import { createShadowRoot } from './util/shadow-root';
 // Minimum width to which the iframeContainer can be resized.
 export const MIN_RESIZE = 280;
 
+/**
+ * Client configuration used to launch the sidebar application.
+ *
+ * This includes the URL for the iframe and configuration to pass to the
+ * application using a config fragment (see {@link addConfigFragment}).
+ *
+ * @typedef {{ sidebarAppUrl: string } & Record<string, unknown>} SidebarConfig
```

Comment:
I'm having trouble discerning the differentiation between `SidebarConfig` and `SidebarContainerConfig` in practice. I can see that `createSidebarIframe` expects a `SidebarConfig` parameter, but in practice it is actually called with `SidebarContainerConfig & SidebarConfig` (because that's what is given to `Sidebar`'s constructor as `config`). Can you help me understand the separation here and/or whether either would ever be used (in practice) independently?

Reply:
`SidebarConfig` is the configuration that is needed to create the sidebar iframe, including the configuration passed to the sidebar application in the URL fragment. `SidebarContainerConfig` is the configuration used by the container of the sidebar iframe.